PythonVisualizations/Stack.py

```diff
@@ -0,0 +1,322 @@
+            return
+
+        self.startAnimations()
+        callEnviron = self.createCallEnvironment()
+        n = self.list.pop()
+
+        # delete the associated display objects
+        items = (n.display_shape, n.display_val)
+
+        # self.moveItemsOffCanvas(items, N, sleepTime=0.02) ## alternative method to moveItemsBy
+        self.moveItemsBy(items, delta=(0, -400), steps=self.CELL_HEIGHT, sleepTime=.01)
+        self.canvas.delete(n.display_shape)
+        self.canvas.delete(n.display_val)
+
+        # draw an index pointing to the last cell
+        if len(self.list) > 0:
+            indexDisplay = self.createIndex(len(self.list) - 1)
+            callEnviron |= set(indexDisplay)
+
+        # Finish animation
+        self.cleanUp(callEnviron)
+
+        return n.val  # returns value displayed in the cell
+
+    # displays the top val of the stack in a small cell on the bottom right of the window
+    def peek(self):
+        callEnviron = self.createCallEnvironment()
+        self.startAnimations()
+
+        # draw an index pointing to the last cell
+        indexDisplay = self.createIndex(len(self.list)-1)
+        callEnviron |= set(indexDisplay)
+
+        # draw output box
+        outputBox = self.canvas.create_rectangle(
+            self.STACK_X0 + self.CELL_WIDTH * 1.5,
+            self.STACK_Y0,
+            self.STACK_X0 + self.CELL_WIDTH * 2.5,
+            self.STACK_Y0 - self.CELL_HEIGHT,
+            fill=self.OPERATIONS_BG)
+        callEnviron.add(outputBox)
+
+        pos = len(self.list) - 1
+
+        # calculate where the value will need to move to
+        outputBoxCoords = self.canvas.coords(outputBox)
+        midOutputBoxY = (outputBoxCoords[3] + outputBoxCoords[1]) // 2
+        midOutputBoxX = (outputBoxCoords[0] + outputBoxCoords[2]) // 2
+
+        # create the value to move to output box
+        valueOutput = self.copyCanvasItem(self.list[pos].display_val)
+        valueList = (valueOutput,)
+        callEnviron.add(valueOutput)
+
+        # move value to output box
+        toPositions = (midOutputBoxX, midOutputBoxY)
+        self.moveItemsTo(valueList, (toPositions,), sleepTime=.02)
+
+        # make the value 25% smaller
+        newSize = (self.VALUE_FONT[0], int(self.VALUE_FONT[1] * .75))
+        self.canvas.itemconfig(valueOutput, font=newSize)
+
+        # Finish animation
+        self.cleanUp(callEnviron)
+
+    # lets user input an int argument that determines max size of the stack
+    def newStack(self, size):
+        del self.list[:]
+        self.size = size
+        self.display()
+
+    def cellCoords(self, cell_index):  # Get bounding rectangle for array cell
+        return (self.STACK_X0 + self.CELL_BORDER,
+                (self.STACK_Y0 - self.CELL_HEIGHT * (cell_index + 1)) + self.CELL_BORDER,
+                self.STACK_X0 + self.CELL_WIDTH - self.CELL_BORDER,
+                self.STACK_Y0 - self.CELL_HEIGHT * cell_index - self.CELL_BORDER)
```

Comment:
Comparing this to the code for Array.py and SimpleSorting.py, I think that the adjustment for the CELL_BORDER should only happen on one corner or the other other of the rectangle, not both.  This "shrinks" the size of the cell  by two border widths instead of one.

Reply:
when we do that the lines in between the cells become thicker. By putting CELL_BORDER in both x's and y's that makes the lines look normal